### PR TITLE
SigLIP loss

### PR DESCRIPTION
**Pull request by gabrieletijunaityte**
## What does this PR do?

Introduce SigLIP loss from https://arxiv.org/abs/2303.15343 

Also fix logging to wandb the best model as `best_val_val` to `best_val`.

## Before submitting

- [x] Did you make sure **title is self-explanatory** and **the description concisely explains the PR**?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **test your PR locally** with `pytest` command?

### CODE REVIEW

File: src/models/components/loss_fns/siglip_loss.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        **kwargs,
+    ) -> torch.Tensor | Dict[str, torch.Tensor]:
+
+        # Normalise inputs
+        eo_mod = F.normalize(eo_mod, dim=-1)
+        text_mod = F.normalize(text_mod, dim=-1)
+
+        # Get logits: (x @ y.T) * scale + bias
+        scale = self.log_scale.exp()
+        logits = (eo_mod @ text_mod.T) * scale
```
Comment (vdplasthijs) [inline src/models/components/loss_fns/siglip_loss.py:51]: Is the minus sign of the definition of Siglip loss (section 3.2 original paper) taken into account here?

**Review by vdplasthijs — COMMENTED** [review]
See Q re minus sign loss

File: src/models/components/loss_fns/siglip_loss.py (hunk 1 above)
Comment (vdplasthijs) [reply src/models/components/loss_fns/siglip_loss.py:51]: Discussed: paper absorbs it in the bias term. Maybe init is opposite sign but parameter is learnable.

**Review by vdplasthijs — APPROVED** [review]
